Makefile

```diff
@@ -1,12 +1,21 @@
 #
-# github.com/docker/cli 
+# github.com/docker/cli
 #
 
+VERSION ?= unknown-version
+GITCOMMIT ?= $(shell git rev-parse --short HEAD 2> /dev/null || true)
+BUILDTIME ?= $(shell date --utc --rfc-3339 ns 2> /dev/null | sed -e 's/ /T/')
+
+LDFLAGS := -X github.com/docker/cli/cli.GitCommit=${GITCOMMIT} \
+	-X github.com/docker/cli/cli.BuildTime=${BUILDTIME} \
+	-X github.com/docker/cli/cli.Version=${VERSION} \
```

Comment:
Actually, you can't `source` within a Makefile. Would have to move the build into a script

I could put the LDFLAGS setup into a sub Makefile and include it instead though (but that can't be as step of a rule)